static/js/message_store.js

```diff
@@ -304,12 +304,10 @@ exports.update_messages = function update_messages(events) {
 
     // If a topic was edited, we re-render the whole view to get any propagated edits
     // to be updated
-    if (topic_edited) {
-        if (!changed_narrow) {
-            home_msg_list.rerender();
-            if (current_msg_list === message_list.narrowed) {
-                message_list.narrowed.rerender();
-            }
+    if (topic_edited && !changed_narrow) {
+        home_msg_list.rerender();
+        if (current_msg_list === message_list.narrowed) {
+            message_list.narrowed.rerender();
```

Comment:
There are at most 2 message lists in existence in the zulip frontend at a time -- the home_msg_list and narrowed_msg_list.